python/lsst/ip/diffim/utils.py

```
@@ -711,3 +732,354 @@ def plotWhisker(results, newWcs):
+    return fig
+
+
+def displayExposure(exposure, showMasks=True, showVariance=False, showPsf=False, showBars=True,
+                    width=8, height=2.5):
+    """Use matplotlib.pyplot.imshow() to display a afw.image.Exposure,
+    including its masks and variance plane and optionally its Psf.
+
+    @see displayMaskedImage() for params not listed here
+
+    @param exposure Exposure to display
+    @param showPsf Display the exposure's Psf
+
+    @return matplotlib.pyplot.figure dispaying the image
+
+    """
+    plt = importMatplotlib()
+    if not plt:
+        return
+
+    fig = displayMaskedImage(exposure.getMaskedImage(), showMasks,
+                             showVariance=not showPsf,
+                             showBars=showBars, width=width, height=height)
+    if showPsf:
+        plt.subplot(1, 3, 3)
+        psfIm = exposure.getPsf().computeImage()
+        bbox = psfIm.getBBox()
+        extent = (bbox.getBeginX(), bbox.getEndX(), bbox.getBeginY(), bbox.getEndY())
+        display2dArray(psfIm.getArray(), title='PSF', showBars=showBars, extent=extent)
+    return fig
+
+
+def displayCutouts(source, exposure, posImage=None, negImage=None, asHeavyFootprint=False, title=''):
+    """Use matplotlib.pyplot.imshow() to display cutouts within up to
+    three afw.image.Exposure's, given by an input SourceRecord.
+
+    @param source afw.table.SourceRecord defining the footprint to extract and display
+    @param exposure afw.image.Exposure from which to extract the cutout to display
+    @param posImage Second exposure from which to extract the cutout to display
+    @param negImage Third exposure from which to extract the cutout to display
+    @param asHeavyFootprint Display the cutouts as
+    afw.detection.HeavyFootprint, with regions outside the
+    footprint removed
+
+    @return matplotlib.pyplot.figure dispaying the image
+
+    """
+    plt = importMatplotlib()
```

Why is it acceptable for matplotlib to not import correctly? Why not just import matplotlib and let it fail?